quint/src/ir/quintIr.ts

```diff
@@ -269,6 +269,8 @@ export interface QuintOpDef extends WithId, WithOptionalTypeAnnotation, WithOpti
   qualifier: OpQualifier
   /** expression to be associated with the definition */
   expr: QuintEx
+  /**  optional depth of the definition, 0 if top-level */
+  depth?: number
```

Comment:
I would prefer to keep the IR really minimal. If we start extending it for various reasons, we may end up with a baroque IR similar to that of TLC, which I would really like to avoid. Can we introduce `depth` somewhere else? Also, when IR objects are constructed, it's hard to predict and manipulate the depth, so it would require an additional pass anyways.

Reply:
I guess Igor's comment answers Thomas' question: it is undefined when the IR is first constructed by the parser, and filled in only when the name collector collects the definition. I realize now that this doesn't ever make it to the actual IR value, only to the lookup table.

My reason to add it to the IR was that 1. it is specific to the `QuintOpDef` type, and 2. it could be useful somewhere else, as we do keep track of `depth` in many visitors. However, I do agree with Igor's point - I'll move it to `Definition` somehow.